plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/internal/core/WatchManager.java

```diff
@@ -124,25 +132,26 @@ public void disconnected() {
 
 		@Override
 		public void error(Throwable err) {
-			Trace.warn("Reconnecting. There was an error watching connection {0}:", err, conn.toString());
+			Trace.warn("Reconnecting. There was an error watching connection {0}: ", err, conn.toString());
 			restart();
 		}
 		
-		private synchronized void restart() {
-			if(state.get() == State.DISCONNECTED) {
-				Trace.debug("Restart called but returning early because already disconnected");
+		private void restart() {
+			if(state.get() == State.STARTING) {
+				Trace.debug("Resturning early from restart.  Already staring for project {0} and kind {1}", project.getName(), kind);
 				return;
 			}
-			state.set(State.DISCONNECTED);
-			Trace.debug("Rescheduling watch job for project {0}", project.getName());
-			startWatch(project, backoff, lastConnect);
+			Trace.debug("Rescheduling watch job for project {0} and kind {1}", project.getName(), kind);
+			startWatch(project, backoff, lastConnect, this);
 		}
-		
+
+
+
 		private class RestartWatchJob extends Job{
 			private IClient client;
 
 			RestartWatchJob(IClient client){
-				super("");
+				super("WatchMananger Job");
```

Comment:
good catch, the missing job name was resulting in quite a few unnamed jobs in the progress view - reported by QE, found out via jstack what it was